feat(components): Add destructive  option for confirmation modal (#633)
GetJobber/atlantis: packages/components/src/ConfirmationModal/ConfirmationModal.mdx

} from "@jobber/components/ConfirmationModal";
```

> Confirmation Modal

Reviewer: ```suggestion
```

Reviewer: (don't need this line, page title is sufficient)

> Default type of confirmation Modal

Reviewer 2: We don't need to say `default` as it is the first one on the page. It is assumed that it is the default.

>           onConfirm={() => alert("✅")}

Reviewer: I'd like to change this sandbox example up a bit:

```
() => {
  const [open, setOpen] = useState(false)
  return (
    <>
      <Button type="secondary" variation="destructive" label="Delete Bob" onClick={() => setOpen(true)} />
      <ConfirmationModal
        title="Delete Bob?"
        message={`Deleting Bob will remove their data from your account for good.`}
        confirmLabel="Delete Bob"
        open={open}
        onConfirm={() => alert('Bob has been deleted')}
        onCancel={() => alert('Bob will not be deleted')}
        onRequestClose={() => setOpen(false)}
        type="destructive"
      />
    </>
  )
}
```

As I think it's a clearer demonstration of the use case this way

Author: Sure thing !

          onConfirm={() => alert("✅")}
          onCancel={() => alert("🙅‍♂️")}
          onRequestClose={() => setOpen(false)}

>           type="work"

Reviewer 2: This can be left off as it is supplied with a default value of `work`

Author: removed

  }}
</Playground>

## Destructive

> If the user is confirming an action that is will destroy, delete, or remove

Reviewer 3: Very minor/nitpicky: Can remove "is" from sentence.  So, `If the user is confirming an action that will destroy, delete, or remove`

Author: I had that doubt too, checked with Grammarly and it gave me okay :P

  }}
</Playground>

## Destructive

Sometimes when the action is critical or destructive in nature (e.g. Deleting

> user or account), you may want to use `destructive type` modal.

Reviewer 2: I don't think that we need to give an example here. (deleting user of account)

Reviewer: ```suggestion
If the user is confirming an action that is will destroy, delete, or remove something, use a destructive ConfirmationModal to visually reinforce the potential consequences.
```

<Playground>
  {() => {
    const [open, setOpen] = useState(false);
    return (
      <>

>         <Button label="Open" onClick={() => setOpen(true)} />

Reviewer: ```suggestion
        <Button label="Delete Bob" variation="destructive" type="secondary" onClick={() => setOpen(true)} />
```
